src/api/api.js

```diff
@@ -36,78 +35,85 @@ api_schema.register_api(require('./func_api'));
 api_schema.register_api(require('./func_node_api'));
 api_schema.compile();
 
-class APIClient {
-
-    constructor(rpc, default_options) {
-        this.rpc = rpc;
-        this.options = _.create(default_options);
-        this.RPC_BUFFERS = RPC_BUFFERS;
-
-        // define the client properties
```

Comment:
The reason I added all these properties explicitly was to allow our typescript compiler be ab le to understand the structure of that client and not give errors in the editor on every access to these properties.

So instead of setting those explicitly, can we just apply a type using jsdoc? You can define the type itself (APIClient) in nb.d.ts and then just make sure the new_client function is defined to `@return {nb.APIClient}`. Let me know if you need more help to get this done.

Reply:
Yes. this is a good idea.
But this will only give you a shallow structure, what really we want is a scheme to typescript definition converter (the same way I implemented for the FE schema). But this is for another task.

Reply:
Shallow is enough to start with

Reply:
Can you add this APIClient type as part of this PR?
I hate to see our editors start marking all our rpc calls in red.

Reply:
I will, as a last thing after I fix the failing tests

Reply:
It errors with strict type checking, here is an example:
```bash
$ cat a.js
'use strict';
function new_client() { return {}; }
const client = new_client();
client.bucket.create_bucket();

$ tsc --allowJs --checkJs --noEmit a.js
<no errors>

$ tsc --allowJs --checkJs --noEmit --strict a.js
a.js:4:8 - error TS2339: Property 'bucket' does not exist on type '{}'.

4 client.bucket.create_bucket();
         ~~~~~~

Found 1 error.
```

Reply:
Your case is not accurate: 
```
'use strict';
const proto = {};
function new_client() { return Object.create(proto); }
const client = new_client();
client.bucket.create_bucket();

tsc --allowJs --checkJs --noEmit a.js
<no errors>

tsc --allowJs --checkJs --noEmit --strict a.js
<no errors>
```

Reply:
But I still see the benefit for auto-complete reasons

Reply:
done